types/node/index.d.ts

```diff
@@ -2833,14 +2833,17 @@ declare module "fs" {
      */
     export function readFileSync(filename: string, options?: { flag?: string; }): Buffer;
     export function writeFile(filename: string, data: any, callback?: (err: NodeJS.ErrnoException) => void): void;
+    export function writeFile(filename: string, data: any, encoding: string, callback?: (err: NodeJS.ErrnoException) => void): void;
```

Comment:
This is what I meant above, there are more issues :o) - it's up to you to decide.
Worst case is that other modules rely on the optional callbacks resulting in test fails. I commented here as you added a new definition which is for sure not used anywhere yet.
If you include it in this PR I would recommend to change the title. There are even more APIs where callback is not mandatory in fs.